src/adagrams.js

```diff
@@ -1,7 +1,97 @@
+const _ = require('lodash');
+
 const Adagrams = {
   drawLetters() {
-    // Implement this method for wave 1
+    const allLetters = ["A", "A", "A", "A", "A", "A", "A", "A", "A", "B", "B", "C", "C", "D", "D", "D", "D", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "E", "F", "F", "G", "G", "G", "H", "H", "I", "I", "I", "I", "I", "I", "I", "I", "I", "J", "K", "L", "L", "L", "L", "M", "M", "N", "N", "N", "N", "N", "N", "O", "O", "O", "O", "O", "O", "O", "O", "P", "P", "Q", "R", "R", "R", "R", "R", "R", "S", "S", "S", "S", "T", "T", "T", "T", "T", "T", "U", "U", "U", "U", "V", "V", "W", "W", "X", "Y", "Y", "Z"];
+
+    //use lodash .sampleSize method to get a random sample of 10 letters 
+    const lettersInHand = _.sampleSize(allLetters, 10);
+
+    return lettersInHand;
+
+  },
+
+  usesAvailableLetters(input, lettersInHand) {
+    input = input.toUpperCase();
+
+    let lettersObject = {};
+
+    lettersInHand.forEach(function (letter) {
+      if (lettersObject[letter] !== undefined) {
+        lettersObject[letter] += 1;
+      } else {
+        lettersObject[letter] = 1;
+      }
+    });
+
+    const inputArray = input.split("");
```

Comment:
Hm, it felt weird that above you basically re-assigned `input` as itself with `input = input.toUpperCase()`. And now here, you make `inputArray`. It might feel better if you inline all of it into one call: `const inputArray = input.toUpperCase().split('');`. Also, I tend to prefer variable names that talk less about the data type and more about meaning, so maybe consider something like `letters` here instead. Lastly: Don't forget our preference for single-quotes ;)